compiler/damlc/tests/daml-test-files/Transferrable.daml

```diff
@@ -0,0 +1,97 @@
+{-# LANGUAGE UndecidableSuperClasses #-}
+{-# LANGUAGE FlexibleContexts #-}
+
+daml 1.2
+module Transferrable where
+
+import DA.Record
+
+class (Template t, Eq t, HasField "issuer" t Party, HasField "owner" t Party) => Issuable t
+
+template Issuable t => Issuance t with
+    assetToIssue: t
+  where
+    signatory assetToIssue.issuer
+
+    controller assetToIssue.owner can
+      AcceptIssuance: ContractId t
+        do create assetToIssue
+
+template (Issuable t, TransferProposalInstance t) => Transferrable t
```

Comment:
I was surprised that I needed the `TransferProposalInstance` constraint. Need to think about this